File: src/LeetCode9IsPalindrome.java
public class LeetCode9IsPalindrome {
    public static void main(String[] args) {
        System.out.println(isPalindrome(10));
    }
    public static boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int reverse = 0;
        int number = x;
        while(x > 0) {
            int temp = x % 10;
            x = x / 10;
            reverse = reverse * 10 + temp;
        }
        System.out.println(number + " " + reverse);
        return number == reverse;
    }
}

File: src/LeetCode9IsPalindrome.kt
fun main(args: Array<String>) {
    println(isPalindrome(121))
}

fun isPalindrome(x: Int): Boolean {
    return if (x < 0) {
        false
    } else {
        var temp = x
        var reverse = 0
        while (temp > 0) {
            reverse = reverse * 10 + temp % 10
            temp /= 10
        }
        reverse == x
    }
}